### ISSUE TITLE
Can provide Normal model?

### ISSUE THREAD

**Issue by bbk40221** [starVisionTeam/SeIF#1]
In train_Normal/demo_Normal.py, the model path is "/media/lx/4A42-E0B2/code_github/SePIFU_08_15into/checkpoints/model_Normal-Net/Train_Normal_epoch_27_1984", 

```python
netG = NormalNet(opt)
        netG.to(cuda)
        # 2023/05/06 we get the 27_1984
        model_path = '/media/lx/4A42-E0B2/code_github/SePIFU_08_15into/checkpoints/model_Normal-Net/Train_Normal_epoch_27_1984'
        netG.load_state_dict(torch.load(model_path, map_location=cuda))
        netG.eval()
```

This model can't be found in repository, can provide this model? 
Thanks.

**Comment by lxzlwhy** [comment]
Hi, because our paper is still under review, we will publish the normal model when the paper is about to be received. Please wait for a while. Thank you for your interest in our work.

**Comment by lxzlwhy** [comment]
Hi, because our paper is still under review, we will publish the normal model when the paper is about to be accepted. Please wait for a while. Thank you for your interest in our work.

**Comment by FernandoConti** [comment]
Hi, is there an update on the Normal model release? Thank you so much in advance!

**Comment by Megidd** [comment]
Any update on it?

**Comment by FernandoConti** [comment]
Hi Megidd, why didn't that GitHub work? Just curious :-)

**Comment by Megidd** [comment]
@FernandoConti I have shown below two normal maps:

* The left one is a sample created by the SEIF normal model.
* The right one is another sample created by https://github.com/brycedrennan/imaginairy-normal-map

As can be observed, the SEIF normal model - left one - creates much more fine details:

![2024-04-29 06_18_05-Skype - Copy](https://github.com/starVisionTeam/SeIF/assets/17475482/c2751d27-b36e-4e1d-acd7-7c4302bc227f)

The right map creates a messed up 3D model eventually, maybe due to the normal map lacking the fine details:

![2024-04-29 06_17_28-Skype - Copy](https://github.com/starVisionTeam/SeIF/assets/17475482/43dcb0d0-0c63-47c3-88a5-6c306dbb3fd4)

**Comment by FernandoConti** [comment]
Thank you so much for showing me! Yes, much less detail and and incorrect normale as well. 

Okay, unfortunately so we're not hearing anything from the creators, maybe we can train a new normal model ourselves. We need a library of highres head scans to train a new model, I'm going to research that.

**Comment by Megidd** [comment]
@FernandoConti That's right.

We believe that it might be possible to just train their normal model in some weeks. Considering the fact that SEIF is apparently using public datasets for their training.

**Comment by FernandoConti** [comment]
@Megidd Yes, I believe that as well. Unfortunately all 3D databases I found need academic approval.

It's really disappointing to see another incomplete 'release' on Github. Especially because it's based on a LOT of other open source software. I would say: if you don't want to share (=make it open source) then DON'T share it and then you also DON'T get the applause. Because they all want the applause without doing the actual work. I wouldn't hire these creators, that's for sure.

Okay well, please keep me updated on your progress, I will do the same!

**Comment by leyuanCv** [comment]
Thank you for your interest in our work. We will upload the trained normal estimation model soon.

**Comment by FernandoConti** [comment]
Hello @Comfyux , thank you for reaching out to me!

The normal estimation model download link has been added to this repo, but it is impossible for us to download it, we cannot access the Baidu network.

Would you maybe be able to help us?

**Comment by nvdomidi** [comment]
@FernandoConti 

I was able to download the file. I've shared it [here](https://drive.google.com/file/d/1wjhKPum42TIgMLKSPtN2ptI03kU1U965/view?usp=sharing) for others who don't have access to baidu.

**Comment by FernandoConti** [comment]
@nvdomidi Thank you so much!

**Comment by Megidd** [comment]
# Approach

We tested by these steps:

1. Cloned this [repo](https://github.com/sama0626/Head_normal/tree/master) on master branch. 
2. Created a `checkpoints` folder in the main directory, then pasted the downloaded normal estimate model `Train_Normal_epoch_27_1984` into that folder.
3. Took our custom image and renamed it to `1.png` and `1_mask.png` and put them into the `image` folder.
4. Ran the `demo.py` and got `1_normal.png`.
5. Pasted `1.png`, `1_mask.png` and `1_normal.png` into SeIF's `demo` folder.
6. Ran `main_test.py`.

# Result

Our sample result is shown below. @FernandoConti What do your results look like?

![SeIF_result1](https://github.com/starVisionTeam/SeIF/assets/17475482/66ddfcf5-c64f-4827-ba5d-524088be6a01)

**Comment by FernandoConti** [comment]
Hi @Megidd, we have similar results, they are pretty bad. Apparently the creators have been cherry picking their results...

**Comment by leyuanCv** [comment]
> Hi @Megidd, we have similar results, they are pretty bad. Apparently the creators have been cherry picking their results...

Could you please share your input images so we can investigate the cause? We have tested up to 25,000 front-facing images without glasses. It is disheartening to receive such feedback.
